integration-tests/test_connection.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+import conftest
+
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_connection(insights_client):
+    """
+    Test insights-client --test-connection
+    `test_connection` method returns 0 if success, 1 if failure
+    """
+    url_test = "End Upload URL Connection Test: SUCCESS"
+    api_test = "End API URL Connection Test: SUCCESS"
+
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    test_connection = insights_client.run("--test-connection")
+    assert url_test in test_connection.stdout
+    assert api_test in test_connection.stdout
+
+
+def test_http_timeout(insights_client):
+    """
+    Test http_timeout config option
+    Set http_timeout to a very low value, run --test-connection and check
+    the set time
+    """
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    insights_client.config.http_timeout = 0.1
+    insights_client.config.save()
+
+    output = insights_client.run("--test-connection", check=False)
+    assert output.returncode == 1
+    assert "read timeout=0.1" in output.stdout
+
+
+def test_noauth_proxy_connection(insights_client, test_config):
+    """
+    Test no-auth proxy option with --test-connection option
+    """
+    url_test = "End Upload URL Connection Test: SUCCESS"
+    api_test = "End API URL Connection Test: SUCCESS"
+
+    insights_client.register()
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    no_auth_proxy: str = (
+        "http://"
+        + test_config.get("noauth_proxy", "host")
+        + ":"
+        + str(test_config.get("noauth_proxy", "port"))
+    )
+    insights_client.config.proxy = no_auth_proxy
+    insights_client.config.save()
+
+    test_connection = insights_client.run("--test-connection")
+    assert url_test in test_connection.stdout
+    assert api_test in test_connection.stdout
+
+
+def test_auth_proxy_connection(insights_client, test_config):
+    """
+    Test auth proxy option with --test-connection
+    """
+    url_test = "End Upload URL Connection Test: SUCCESS"
+    api_test = "End API URL Connection Test: SUCCESS"
+
+    insights_client.register()
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    auth_proxy: str = (
+        "http://"
+        + test_config.get("auth_proxy", "username")
+        + ":"
+        + test_config.get("auth_proxy", "password")
+        + "@"
+        + test_config.get("auth_proxy", "host")
+        + ":"
+        + str(test_config.get("auth_proxy", "port"))
+    )
+    insights_client.config.proxy = auth_proxy
+    insights_client.config.save()
+    test_connection = insights_client.run("--test-connection")
+    assert url_test in test_connection.stdout
+    assert api_test in test_connection.stdout
+
+
+def test_url_connection(insights_client):
```

Comment:
since this test is about a wrong `base_url`, please tweak the test name so it mentions that explicitly